drivers/kw41zrf/kw41zrf.c

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (C) 2017 SKF AB
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_kw41zrf
+ * @{
+ * @file
+ * @brief       Basic functionality of kw41zrf driver
+ *
+ * @author      Joakim Nohlgård <joakim.nohlgard@eistec.se>
+ * @}
+ */
+#include <stdint.h>
+#include <string.h>
+#include <errno.h>
+
+#include "log.h"
+#include "msg.h"
+#include "luid.h"
+#include "net/gnrc.h"
+#include "net/ieee802154.h"
+
+#include "kw41zrf.h"
+#include "kw41zrf_netdev.h"
+#include "kw41zrf_getset.h"
+#include "kw41zrf_intern.h"
+#include "vendor/XCVR/MKW41Z4/fsl_xcvr.h"
+#include "vendor/XCVR/MKW41Z4/ifr_radio.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+static void kw41zrf_set_address(kw41zrf_t *dev)
+{
+    DEBUG("[kw41zrf] Set MAC address\n");
+    eui64_t addr_long;
+    /* get an 8-byte unique ID to use as hardware address */
+    luid_get(addr_long.uint8, IEEE802154_LONG_ADDRESS_LEN);
+    /* make sure we mark the address as non-multicast and not globally unique */
+    addr_long.uint8[0] &= ~(0x01);
+    addr_long.uint8[0] |=  (0x02);
+    /* set short and long address */
+    kw41zrf_set_addr_long(dev, ntohll(addr_long.uint64.u64));
```

Comment:
You are calling `byteorder_swapll` in `kw41zrf_set_addr_long()`, so that undoes your `ntohll()`.

Reply:
It's because `netdev_set()` needs a swap and `kw41zrf_set_addr_long()` is the cleanest place to put it. I guess whether we swap here is actually arbitrary except that the unique bits between sequential chips arrive in better locations (that's subjective) with this swap.

Reply:
it's just that there was some confusion with the order of the bytes in the `at86rf2xx` driver in the past.
So we ended up with just always using `eui64_t` for the address to have a defined order.